File: temis-uvi.jl
# Analysis of interannual variability of incident solar UV index radiation from TEMIS data.

# Original data source and documentation:
# https://www.temis.nl/uvradiation/product/uvi-uvd.html

# Guided by:
# https://towardsdatascience.com/read-csv-to-data-frame-in-julia-programming-lang-77f3d0081c14

println("aqui começa")

using CSV
using DataFrames

uvi_path = "/home/santiago/Projetos/temis-uvi/TEMIS_UVI_Acarau.csv"
uvi_file = CSV.File(uvi_path, header=1, delim=";")
uvi_data = DataFrame(uvi_file)

# CSV file column names:
# yyyy; mm; dd; date; uvief; uvdef; uvdvf; uvddf; ozone

#println(typeof(temis_f))
#for row in temis_f 
#    println("Valores: $(row.yyyy), $(row.mm), $(row.dd), $(row.uvief),  $(row.uvdef), $(row.uvdvf), $(row.uvddf), $(row.ozone)")
#    println("Valores: $(row.yyyy), $(row.mm), $(row.dd), $(row.ozone)")
#end

#println(uvi_file[1,1]) 

mi = 1
for row in uvi_file
    println("$(mi) | Date: $(row.date) | Ozone (DU): $(row.ozone)")
    local mi = mi + 1
end

# println(uvi_file.ozone[])

println("aqui termina")

File: temis-uvi.py
# Analysis of interannual variability of incident solar UV index radiation from TEMIS data.

# Original data source and documentation:
# https://www.temis.nl/uvradiation/product/uvi-uvd.html

# Instructions from:
# https://realpython.com/python-csv/

#import csv

print('começa aqui...')

import numpy as np
uvi_path = '/home/santiago/Projetos/temis-uvi/TEMIS_UVI_Acarau.csv'
uvi_data = np.genfromtxt(uvi_path, delimiter=';', names=True, dtype=None, encoding=None)
print(uvi_data.dtype.names)

yy = uvi_data['yyyy']
mm = uvi_data['mm']
dd = uvi_data['dd']
dt = uvi_data['date']
ief = uvi_data['uvief']
uef = uvi_data['uvdef']
dvf = uvi_data['uvdvf']
ddf = uvi_data['uvddf']
ozo = uvi_data['ozone']

clim_ozo = []
for m in range(1,13):
    print(m)
    print(np.average(ozo[np.where(mm == m)]))
    np.append(clim_ozo, np.average(ozo[np.where(mm == m)]))
#    clim_ozo = np.average(ozo[np.where(mm == m)])


print(clim_ozo)

ijan = np.where(mm == 1)

print(ijan)

print(ozo[ijan])

print(np.average(ozo[ijan]))

print(range(1,12))
range(12)

for i in range(1, 12):
    print(i)

print('chegamos ao fim!')

